packages/mobile/src/hooks/useLocalTrackImage.ts

```diff
@@ -0,0 +1,53 @@
+import { useState, useCallback, useMemo } from 'react'
+
+import { SquareSizes } from '@audius/common'
+import type { ImageSourcePropType, ImageURISource } from 'react-native'
+
+import { getLocalCoverArtPath } from 'app/services/offline-downloader'
+
+export type LocalImageSource = {
+  source: ImageSourcePropType | null
+  handleError: () => void
+}
+
+export const useLocalTrackImage = (
+  trackId: string | undefined
+): LocalImageSource => {
+  const [imageSourceIndex, setImageSourceIndex] = useState(0)
+  const [failedToLoad, setFailedToLoad] = useState(false)
+  const imageSources = useMemo(
+    () =>
+      trackId
+        ? Object.values(SquareSizes)
+            .reverse()
+            .map(
+              (size): ImageURISource => ({
+                uri: getLocalCoverArtPath(trackId, size.toString()),
+                width: parseInt(size.split('x')[0]),
+                height: parseInt(size.split('x')[1])
+              })
+            )
+        : [],
+    [trackId]
+  )
+
+  const handleError = useCallback(() => {
```

Comment:
I don't think we need this error handler or the imageSourceIndex in this case because the result of this hook is an array of sizes which the Image component can accept and automatically select the correct size

Reply:
Ok, I tried doing this and realized I'm unable to get it to work with both the remote and local fetching. Even with a hard-coded local source it's sometimes showing the blank image... I might need to pair on this a bit with someone who knows how `DynamicImage` and `ImageLoader` work